deploy/cr.yaml

```diff
@@ -19,5 +19,6 @@ spec:
     type: "curator"
     schedule: "30 3 * * *"
   collection:
-    type: "fluentd"
-    nodeSelector: "logging-infra-fluentd=true"
+    logCollection:
+      type: "fluentd"
+      nodeSelector: "logging-infra-fluentd=true"
```

Comment:
How will this work for rsyslog?  Will we need to change the cr to use `type: rsyslog` and `nodeSelector: "logging-infra-rsyslog=true"`?
Would it be possible to have a map or list here e.g.
```
     logCollection: 
       - type: "fluentd"
         nodeSelector: "logging-infra-fluentd=true"
       - type: "rsyslog"
         nodeSelector: "logging-infra-rsyslog=true"
```

Reply:
Yes, it would instead be `type: "rsyslog"`. Would we want to support both fluentd and rsyslog running at the same time?

Reply:
Not sure - how do I change this to deploy rsyslog instead of fluentd?

Reply:
Maybe we should consider changing to something like `logging-infra-collector=[fluentd|rsyslog]` or `logging-infra=<componentname>` for flexibility

Reply:
this is user defined in this case. so they could update it to match whatever they would prefer but we could have a different default value for each component.

Reply:
@richm this cr is actually missing `nodeSelector` being under `fluentd:` to match the log collection spec.
```
logCollection:
  type: "fluentd"
  fluentd:
    nodeSelector: "logging-infra-fluentd=true"
```
I'll update this.